backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Controllers/UserController.cs

```diff
@@ -1,30 +1,35 @@
-﻿using EasyMeets.Core.BLL.Interfaces;
+﻿using System.Security.Claims;
+using EasyMeets.Core.BLL.Interfaces;
 using EasyMeets.Core.Common.DTO.User;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace EasyMeets.Core.WebAPI.Controllers
 {
+    [Authorize]
     [ApiController]
     [Route("[controller]")]
     public class UserController : ControllerBase
     {
         private readonly IUserService _userService;
-        public UserController(IUserService userService)
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        public UserController(IUserService userService, IHttpContextAccessor httpContextAccessor)
         {
             _userService = userService;
+            _httpContextAccessor = httpContextAccessor;
         }
-
+        
         [HttpGet("current/{id}")]
         public async Task<ActionResult<UserDto>> GetCurrentUserByIdAsync(int id)
         {
-            var availabilitySlots = await _userService.GetCurrentUserAsync(id);
+            var availabilitySlots = await _userService.GetCurrentUserAsync(id, GetCurrentUserEmail());
```

Comment:
It's the wrong way to do it.
You need to change the backend to get identity from user httpContext. And the frontend is supposed not to send user id for these kinds of routes.

Reply:
I changed endpoint to 'user/current'. We don't have Id there anymore. Current user gets from database by email that we receive from HttpContext

Reply:
Do it for GetUserPreferences also. And change frontend, because it will be broken

Reply:
That endpoint deleted at all because it did the same thing as GetCurrentUser